scripts/buildWebtypes.js

```diff
@@ -0,0 +1,254 @@
+const path = require('path');
+const fs = require('fs');
+const os = require('os');
+const { execSync } = require('child_process');
+
+const PLAIN_WEB_TYPES_FILE = 'web-types.json';
+const LIT_WEB_TYPES_FILE = 'web-types.lit.json';
+
+const API_DOCS_BASE_PATH = 'https://cdn.vaadin.com/vaadin-web-components';
+
+const blacklistedPackages = [
+  /^vaadin-/,
+  /^component-base/,
+  /^field-base/,
+  /^field-highlighter/,
+  /^icons/,
+  /^input-container/,
+  /^lit-renderer/,
+  /^polymer-legacy-adapter/,
+];
+
+// Additional attributes that will be added to all components
+const additionalAttributes = [
+  // Theme attribute is not properly declared at the moment
+  {
+    name: 'theme',
+    description: 'The theme variants to apply to the component.',
+    type: 'string | null | undefined',
+  },
+];
+
+/**
+ * Get packages using lerna, excluding blacklisted packages
+ */
+function getRelevantPackages() {
+  const output = execSync('./node_modules/.bin/lerna ls --json --loglevel silent');
+  const allPackages = JSON.parse(output.toString()).map((project) => project.name.replace('@vaadin/', ''));
+  return allPackages.filter((pkg) => !blacklistedPackages.some((blacklistedPackage) => pkg.match(blacklistedPackage)));
+}
+
+function loadAnalysis() {
+  const analysisPath = path.resolve('./analysis.json');
+  try {
+    return JSON.parse(fs.readFileSync(analysisPath, 'utf8'));
+  } catch (e) {
+    throw new Error(
+      `Could not read output of the Polymer Analyzer from: ${analysisPath}. Make sure to run the Polymer Analyzer before generating web-types.`,
+    );
+  }
+}
+
+/**
+ * Maps a type string such as `'boolean | null | undefined'` into an array of type
+ * strings, such as `['boolean', 'null', 'undefined']`
+ * @param typeString
+ * @returns {string[]}
+ */
+function mapType(typeString) {
+  const sanitizedTypeString = (typeString || '').replace(/[!()]/g, '');
+  const types = sanitizedTypeString.split('|');
+  return types.map((type) => type.trim());
+}
+
+/**
+ * Transforms a description text of an element, attribute, property, event to:
+ * - make relative docs links point to the absolute documentation URL
+ * @param packageJson
+ * @param description
+ * @returns {*}
+ */
+function transformDescription(packageJson, description) {
+  // Transform relative documentation links to absolute ones
+  description = description.replace(
+    /\(#\/elements\/(.*)\)/g, // Matches "(#/elements/$1)"
+    `(${API_DOCS_BASE_PATH}/${packageJson.version}/#/elements/$1)`,
+  );
```

Comment:
> Instead of passing the "description" data as is, please update it to fix relative links that start with #/elements/ to add base path e.g. https://cdn.vaadin.com/vaadin-web-components/23.2.0-alpha3/ so that they can be opened.

Added a `transformDescription` function that should be called for every description. This part looks for markdown links, and replaces them with an absolute URL to the API docs, and the respective version from package.json.